controllers/user/profile.go

```diff
@@ -28,7 +28,7 @@ func UserProfileHandler(c *gin.Context) {
 	Ts, _ := publicSettings.GetTfuncAndLanguageFromRequest(c)
 	messages := msg.GetMessages(c)
 
-	if id == 0 && ContainsNonNumbersChars(c.Param("id")) {
+	if c.Param("id") != "0" && id == 0 && ContainsNonNumbersChars(c.Param("id")) {
```

Comment:
If c.Param("id") is not "0" and not a number, then id = 0. So you don't have to check for ContainsNonNumbersChars.
The reason is simple, if you convert something else than a number to int, the variable is equal to 0.
So id is equal to 0 if c.Param("id") = 0 or if c.Param("id") is not a number

Reply:
The point of this condition is to send users to username search if the write /user/xxx , (which will make id into 0), but user/0 is the anonymous profile (also id 0) so i also have to check if it contains letters

Reply:
I understand :) and I think if you only do this `if c.Param("id") != "0" && id == 0` instead of this `if c.Param("id") != "0" && id == 0 && ContainsNonNumbersChars(c.Param("id"))`, you should get the same result. (explanations above)